synapse/tests/test_lib_ast.py

```diff
@@ -359,9 +359,28 @@ async def test_ast_variable_props(self):
             # Sad paths
             q = '[test:str=newp -.newp]'
             await self.asyncraises(s_exc.NoSuchProp, core.nodes(q))
+
             q = '$newp=newp [test:str=newp -.$newp]'
             await self.asyncraises(s_exc.NoSuchProp, core.nodes(q))
 
+            q = '$newp=(foo, bar) [test:str=newp] $lib.print(:$newp)'
+            await self.asyncraises(s_exc.NoSuchProp, core.nodes(q))
```

Comment:
Did you consider raising a different exception in these cases?  Feels like it's some sort of a bad type/value instead of NoSuchProp.

Reply:
We'd want to add `isinstance` checks to the sad path logic to make that work right. Def don't want to do that inline for the happy path if we can avoid it.

Reply:
Updated to raise a more informative StormRuntimeError in most cases